examples/github_search/lib/base/app/github_search.dart

```diff
@@ -65,7 +67,7 @@ class __MyMaterialAppState extends State<_MyMaterialApp> {
       await safeRun(
           () => FirebaseMessaging.instance.getToken(vapidKey: webVapidKey));
     }
-    FirebaseMessaging.instance
+    await FirebaseMessaging.instance
```

Comment:
If we wait for the result of the initial message, why the `then` method is used ?

```
final initialMessage = await FirebaseMessaging.instance.getInitialMessage()
onInitialMessageOpened(initialMessage);
```

or just
`onInitialMessageOpened(await FirebaseMessaging.instance.getInitialMessage());`

Reply:
Your proposed first solution will work with slight modifications due to an [async gap](https://www.youtube.com/watch?v=bzWaMpD1LHY):

https://user-images.githubusercontent.com/61236867/181473340-179360da-f6d3-4d9b-a133-16923067987f.mov